### PR TITLE
Add the faiss extension

### PR DESCRIPTION
**Pull request by JAicewizard**
This adds the faiss extension, and mirrors the custom-vcpkg changes from the CI tools repo. All tests pass (except docs, since the extension is not actually uploaded) in my fork, hope they pass here too :)

### CODE REVIEW

File: extensions/faiss/description.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  vcpkg_url: "https://github.com/jaicewizard/vcpkg.git"
+  vcpkg_commit: "4973a339118ad0e2455a7dad1348baa8c27bce16"
+
+repo:
+  github: arjenpdevries/faiss
+  ref: ac09d01af7ce9711f74097012f4bff4b5275147f
+
+docs:
+  hello_world: |
+    -- github.com/arjenpdevries/faiss/blob/main/README.md
```
Comment (carlopi) [inline extensions/faiss/description.yml:22]: This is generally a SQL-only demo of the extension, you can have a look for example at how it's rendered here for h3: https://community-extensions.duckdb.org/extensions/h3.html#example.

That said, SQL comment is valid SQL, +1

File: extensions/faiss/description.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  extended_description: |
+    The FAISS extension allows duckdb users to store vector data in faiss, and query this data, making relable vector search more accessable.
```
Comment (carlopi) [inline extensions/faiss/description.yml:24]:
```suggestion
    The FAISS extension allows duckdb users to store vector data in faiss, and query this data, making reliable vector search more accessible.
```
Also, fine changing / iterating on this later.

**Review by carlopi — APPROVED** [review]
Thanks! I was curious what were the hurdles Wasm side would be, I guess a few, but maybe there is a path?

File: extensions/faiss/description.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+extension:
+  name: faiss
+  description: Provides a subset of the faiss API to duckdb
+  version: 0.9.0
+  language: C++
+  build: cmake
+  license: MIT
+  maintainers:
+    - Jaap Aarts, Arjen P de Vries
```
[the remark this answers is not in the document]
Comment (JAicewizard) [reply extensions/faiss/description.yml:9]: ah Yeah I see it is a YAML list now, not sure how I missed that. I will fix that

**Comment by carlopi** [comment]
Welcome!

**Pull request merged**